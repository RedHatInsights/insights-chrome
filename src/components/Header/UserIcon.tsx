import React, { useContext, useEffect, useState } from 'react';
import { Avatar } from '@patternfly/react-core/dist/dynamic/components/Avatar';

import ImgAvatar from '../../../static/images/img_avatar.svg';
import ChromeAuthContext from '../../auth/ChromeAuthContext';

const UserIcon = () => {
  const auth = useContext(ChromeAuthContext);
  const [avatar, setAvatar] = useState(ImgAvatar);

  const getImage = (img: HTMLImageElement) => {
    if (img.width === 140) {
      setAvatar(img.src);
    }
  };

  useEffect(() => {
    const img = new Image();
    img.src = `https://access.redhat.com/api/users/avatar/${auth.user.identity.user?.username ?? ''}/`;
    img.onload = () => getImage(img);
  }, []);

  return <Avatar src={avatar} alt="User Avatar" />;
};

export default UserIcon;
