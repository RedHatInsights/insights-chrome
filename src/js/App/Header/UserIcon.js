import React, { Component } from 'react';
import { Avatar } from '@patternfly/react-core4/dist/js/components/Avatar/Avatar';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export class UserIcon extends Component {

    constructor(props) {
        super(props);
        this.state = {
            account: this.props.account,
            avatar: 'apps/chrome/assets/images/img_avatar.svg'
        };
    }

    getImage = (img) => {
        if (img.width === 140) {
            this.setState({ avatar: img.src });
        }
    }

    componentDidMount() {
        const img = new Image();
        img.src = `https://access.redhat.com/api/users/avatar/${this.state.account.username}/`;
        img.onload = (() => this.getImage(img));
    }

    render() {
        const { avatar } = this.state;

        return (
            <Avatar src={ avatar } alt='User Avatar'/>
        );
    }
}

UserIcon.propTypes = {
    account: PropTypes.shape({
        username: PropTypes.string
    })
};

export default connect(({ chrome: { user: { identity: { user: { username } } } } }) => ({
    account: {
        username: username
    }
}))(UserIcon);
