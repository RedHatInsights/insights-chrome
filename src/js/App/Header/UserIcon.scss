.ins-c-dropdown-icon {
    height: 36px;
    width: 36px;
}