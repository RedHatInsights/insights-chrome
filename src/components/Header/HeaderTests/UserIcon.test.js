import React from 'react';
import UserIcon from '../UserIcon';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

describe('<UserIcon />', () => {
  let initialState;
  let mockStore;
  beforeEach(() => {
    mockStore = configureStore();
    initialState = {
      chrome: {
        user: {
          identity: {
            user: {
              username: 'test-user',
            },
          },
        },
      },
    };
  });

  it('should render correctly with initial state', () => {
    const store = mockStore(initialState);
    const { container } = render(
      <Provider store={store}>
        <UserIcon />
      </Provider>
    );
    expect(container.querySelector('img')).toMatchSnapshot();
  });
});
