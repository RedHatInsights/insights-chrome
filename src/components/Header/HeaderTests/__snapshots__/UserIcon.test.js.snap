// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserIcon /> should render correctly with initial state 1`] = `
<img
  alt="User Avatar"
  class="pf-v5-c-avatar"
  src="test-file-stub"
/>
`;
